### ISSUE TITLE
Unexpected Completion Behavior

### ISSUE THREAD

**Issue by tschwaller** [google/python-fire#148]
Consider the following simple example (from the docs)

```
#!/usr/bin/env python

import fire

def add(x, y):
  return x + y

if __name__ == '__main__':
  fire.Fire(add, name='./add.py')
```

The first command completion shows `add` and the second immediately `--x  --y`. 

This is confusing, since I would expect to see `X` and after I entered the first number I would expect `Y` or or even `X Y` at the very beginning so I know I have to enter 2 numbers, especially since the Usage shows

```
Usage:       ./add.py X Y
             ./add.py --x X --y Y
```

As far as I can see it is also not possible to distinguish mandatory from optional arguments. In the case above `X` and `Y`are mandatory and I would not expect to see `--x`and  `--y`. An optional argument could e.g. be `--modulo`, which would have a default value if not used, but how to distinguish that in the Python code from the other arguments?

**Comment by dbieber** [comment]
Good point about the order, thanks for bringing this up.
As for mandatory vs optional args, optional args (those with default values) should be shown in square brackets in the help screen.
We're actively working on improving the help and usage screens. The tab completion also has room for improvement, but isn't actively being worked on.

**Comment by tschwaller** [comment]
Thanks for input. Ah, yes, something like

`def add(x, y=0):`

shows

```
Usage:       ./add.py X [Y]
             ./add.py --x X [--y Y]
```

and should actually not show `--x X` since this is mandatory. Although the command completion might look like a minor feature, it is actually one of the killer features of Fire since it works out of the box (#32, #98 will make that even better when implemented). The more natural the completion, the better! 👍 

P.S. `./add.py -- --help` is also kind of unnatural. There should be an exception for that, so you could always say e.g. `./add.py --help` and `./add.py add --help`which would give the commands and its completions a very natural behavior. just my 2 cents.

**Comment by dbieber** [comment]
> P.S. `./add.py -- --help` is also kind of unnatural. 

Yes! We now support just ./add.py --help as well (and as of a few commits ago, ./add.py -h as well). Now the only time this won't work is if the function has a parameter named `help`.

**Comment by stefaang** [comment]
Found another minor one: when using classes, the completion should not suggest the `self` attribute.

**Comment by bluedistro** [comment]
> Found another minor one: when using classes, the completion should not suggest the `self` argument.

I think this issue is resolved
